pr_etiquette.md

```diff
@@ -0,0 +1,62 @@
+# Etiquette for pull requests
+
+## As an author
+
+Submitting pull requests in EDC should be done while adhering to a couple of simple rules.
+
+- Familiarize yourself with [coding style](styleguide.md), [architectural patterns](docs/architecture-principles.md) and
+  other contribution guidelines.
+- No surprise PRs please. Before you submit a PR, open a discussion or an issue outlining your planned work and give
+  people time to comment. It may even be advisable to contact committers using the `@mention` feature. Unsolicited PRs
+  may get ignored or rejected.
+- Create focused PRs: your work should be focused on one particular feature or bug. Do not create broad-scoped PRs that
+  solve multiple issues as reviewers may reject those PR bombs outright.
+- Provide a clear description and motivation in the PR description in Github. This makes the reviewer's life much
+  easier. It is also helpful to outline the broad changes that were made, e.g. "Changes the schema of XYZ-Entity:
+  the `age` field changed from `long` to `String`"
+- If you introduce new 3rd party dependencies, be sure to note them in the PR description and explain why they are
+  necessary
+- Stick to the established code style, please refer to the [styleguide document](styleguide.md).
+- All tests should be green, especially when your PR is in `"Ready for review"`
+- Mark PRs as `"Ready for review"` only when you're prepared to defend your work. By that time you have completed your
+  work and shouldn't need to push any more commits other than to incorporate review comments.
+- Merge conflicts should be resolved by squashing all commits on the PR branch, rebasing onto `main` and
+  force-pushing. Do this when your PR is ready to review.
```

Comment:
Commit messages like `"small improvement"` or `"added test"` are of little value in the log, but if you prefer to resolve merge conflicts on multiple commits as opposed to on a one or two on your fork, that's fine. Just know that upon merging the PR, we always perform a "Squash and merge" anyways.

Reply:
Thanks for you response :-) Yes, the "Squash and merge" at the end is also in my eyes a very good practise, because it keeps the upstream branch history clean.

I overlook the word `should` in the sentence. It's good, that we don't "enforce" this practise at this point. I'm fine with it 😉

Reply:
Thanks for you response :-) Yes, the "Squash and merge" at the end is also in my eyes a very good practise, because it keeps the upstream branch history clean.

I overlook the word should in the sentence. It's good, that we don't "enforce" this practise at this point. I'm fine with it 😉

(sorry for using the wrong user - was in the wrong browser session...)